subprojects/core/src/test/groovy/org/gradle/api/internal/file/archive/TarCopyActionSpec.groovy

```diff
@@ -118,11 +118,7 @@ public class TarCopyActionSpec extends Specification {
         action.execute(new CopyActionProcessingStream() {
             public void process(CopyActionProcessingStreamAction action) {
                 for (FileCopyDetailsInternal f : files) {
-                    if (f.isDirectory()) {
-                        action.processFile(f);
-                    } else {
-                        action.processFile(f);
-                    }
+                    action.processFile(f);
```

Comment:
This logic is correct assuming that `f.isDirectory()` has no side effects. I checked all implementors of `FileCopyDetails.isDirectory()` and that seems to be correct.